source/ShipyardPanel.cpp

```diff
@@ -348,19 +348,56 @@ void ShipyardPanel::Sell(bool toStorage)
 	{
 		auto it = playerShips.begin();
 		message += (*it++)->Name() + ",\n";
-		for(int i = 1; i < MAX_LIST - 1; ++i)
+		for(int i = 1; i < MAX_SHIP_LIST - 1; ++i)
 			message += (*it++)->Name() + ",\n";
 
-		message += "and " + to_string(count - (MAX_LIST - 1)) + " other ships";
+		message += "and " + to_string(count - (MAX_SHIP_LIST - 1)) + " other ships";
 	}
+	map<const Outfit*, int> uniqueOutfits;
 	// To allow calculating the sale price of all the ships in the list,
 	// temporarily copy into a shared_ptr vector:
 	vector<shared_ptr<Ship>> toSell;
-	for(const auto &it : playerShips)
-		toSell.push_back(it->shared_from_this());
+	for(const auto &ship : playerShips)
+	{
+		for(const auto &it : ship->Outfits())
+			if(it.first->Attributes().Get("unique") > 0)
+				uniqueOutfits[it.first] += it.second;
+		toSell.push_back(ship->shared_from_this());
+	}
+	bool hasUniques = !uniqueOutfits.empty();
+	if(hasUniques)
+	{
+		const int uniquesSize = uniqueOutfits.size();
+		int detailedOutfitList = 5;
+		if(detailedOutfitList > uniquesSize)
+			detailedOutfitList = uniquesSize;
+		else
+		{
+			// The less ships we list, the more place we have for unique outfits.
+			if(initialCount < MAX_SHIP_LIST)
+				detailedOutfitList += MAX_SHIP_LIST - initialCount;
```

Comment:
detailedOutfitList might exceed MAX_SHIP_LIST here no?

Reply:
Yes, but it shouldn't exceed 25, MAX_SHIP_LIST + minimal guaranteed detailedOutfitList.